src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonNumberHandling.cs

```diff
@@ -25,7 +27,10 @@ public enum JsonNumberHandling
         /// <summary>
         /// The "NaN", "Infinity", and "-Infinity" <see cref="JsonTokenType.String"/> tokens can be read as
         /// floating-point constants, and the <see cref="float"/> and <see cref="double"/> values for these
-        /// constants will be written as their corresponding JSON string representations.
+        /// constants (such as <see cref="float.PositiveInfinity"/> and <see cref="double.NaN"/>)
+        /// will be written as their corresponding JSON string representations.
+        /// Strings that have escaped characters will be unescaped before reading.
+        /// Leading or trailing trivia, including whitespace, is not allowed.
```

Comment:
So, with such a setting, "\u0031" will get parsed as the integer number 1. Neat.

Reply:
Yes, when `AllowReadingFromString` is active (these comments are for `AllowNamedFloatingPointLiterals`).